bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -730,20 +769,36 @@ private boolean handleEqualizerCommands(String channelId, Command command, Conne
 
     private void startTextToSpeech(Connection connection, Device device, String text)
             throws IOException, URISyntaxException {
+        Integer volume = null;
         if (textToSpeechVolume != 0) {
-            @Nullable
-            ScheduledFuture<?> oldIgnoreVolumeChange = this.ignoreVolumeChange;
-            if (oldIgnoreVolumeChange != null) {
-                oldIgnoreVolumeChange.cancel(false);
-            }
-            this.ignoreVolumeChange = scheduler.schedule(this::stopIgnoreVolumeChange, 2000, TimeUnit.MILLISECONDS);
+            startIgnoreVolumeChange();
+            volume = textToSpeechVolume;
         }
         if (text.startsWith("<speak>") && text.endsWith("</speak>")) {
             String bodyText = text.replaceAll("<[^>]+>", "");
-            connection.sendAnnouncement(device, text, bodyText, null, textToSpeechVolume, lastKnownVolume);
+            connection.sendAnnouncement(device, text, bodyText, null, volume, lastKnownVolume);
         } else {
-            connection.textToSpeech(device, text, textToSpeechVolume, lastKnownVolume);
+            connection.textToSpeech(device, text, volume, lastKnownVolume);
+        }
+    }
+
+    @Override
+    public void startAnnouncment(Device device, String text, String bodyText, String title, @Nullable Integer volume)
+            throws IOException, URISyntaxException {
+        Connection connection = this.findConnection();
+        if (connection == null) {
+            return;
+        }
+        if (volume == null && textToSpeechVolume != 0) {
+            volume = textToSpeechVolume;
+        }
+        if (volume != null && volume < 0) {
```

Comment:
You overwrite what you set three lines before. Is that intended?

Reply:
Yes, it happens only for negative values specified which means 'do not use'. In this case the api requires a null

Reply:
I have added a source comment now

Reply:
This would be sufficient then: `textToSpeechVolume > 0`. Then the second if can be a `else if`. Right?